src/view/Layout.js

```diff
@@ -81,24 +82,52 @@ class Layout extends React.Component {
     }
 
     render() {
-        let childComponents = [];
+        let tabSetComponents = [];
+        let tabComponents = {};
+        let splitterComponents = [];
+
         this.model._layout(this.rect);
-        this.renderChildren(this.model.getRoot(), childComponents);
-        return <div ref="self" className="flexlayout__layout">{childComponents}</div>;
+        this.renderChildren(this.model.getRoot(), tabSetComponents, tabComponents, splitterComponents);
+
+        const nextToIds = [];
```

Comment:
I would like to remove the inner loop of .find() by changing the code to:

        const nextTopIds = [];
        const nextTopIdsMap = {};

        // Keep any previous tabs in the same DOM order as before, removing any that have been deleted
        this.tabIds.forEach(t => {
            if (tabComponents[t]) {
                nextTopIds.push(t);
                nextTopIdsMap[t] = t;
            }
        });
        this.tabIds = nextTopIds;

        // Add tabs that have been added to the DOM
        Object.keys(tabComponents).forEach(t => {
            if (!nextTopIdsMap[t]) {
                this.tabIds.push(t);
            }
        });